Convert all existing documents to be no more than 2 levels

Charrondev opened the pull request:
This PR takes any existing 3rd/4th level documents and

- Merges most 3rd level documents into their existing 2nd level document.
- Some 3rd level documents were moved up to be inside of a 1st level document and become 2nd level documents. Eg. `help/addons/social/facebook -> /help/sso/facebook`.
- Gives them appropriate headings so the fit the structure of the document and can be navigated to easily using the jump-to scrolling.
- Migrates all existing aliases.
- Creates new aliases along with the existing ones.

There was 1 particular weird setup where the embedding docs were trying to be between the developer docs an the help docs. They really are 2 separate docs and were messing with my menu, so instead of nested "Advanced Embedding", that doc has been pulled up and placed on the developer side. I've crosslinked it from the help embedding doc.

slafleche commented on content/developer/embedding/index.md:
```diff
@@ -3,46 +3,125 @@ title: Embedding
-## Embedding Forum
+## Overview
 
-Embedding Vanilla Forums works by putting a snippet of code on your website page, and may be a choice if you want to insert your forum into a site quickly. The snippet points to some javascript and the forum is embedded into the page using an iframe
```
Looks like you've replaced the contents of the embedding page with the advanced page. It should be appended, not deleted.

Charrondev replied:
Handled and by todd's request, the combined page is in the help section

slafleche commented on content/help/addons/groups/managing-groups.md:
```diff
@@ -1,28 +0,0 @@
-  help:
-    parent: addons-groups
-aliases:
-- /addons/groups/managing-groups
```
This alias is missing from the new page.

Charrondev replied:
Fixed

slafleche commented on content/developer/embedding/index.md:
```diff
@@ -3,46 +3,125 @@ title: Embedding
+    identifier: embedding-developer
 aliases:
-- /features/embedding
-- /help/forum-settings/embedding
```
Keep old aliases

Charrondev replied:
Fixed

slafleche commented on content/help/addons/social/index.md:
```diff
@@ -1,16 +0,0 @@
-
-## Social Connect Plugins
-
-Vanilla Forums integrates with a number of social applications that you can use as SSO solutions. Learn more about configuring these applications to work with your forum in this section. 
```
Was this info added anywhere? I don't see an alias either.

slafleche replied:
I would probably keep this page and append all the social integrations inside of it instead of having them as separate pages.

Charrondev replied:
Fixed

slafleche commented on content/help/addons/vanilla-pop/site-hub.md:
```diff
@@ -1,75 +0,0 @@
-tags:
-- Features
-- Vanilla Pop
-- Hub
```
Missing tag in merged page

Charrondev replied:
Fixed

slafleche commented on content/help/ideation/index.md:
```diff
@@ -3,15 +3,13 @@ title: Ideation
-
 Ideation gives forum users the ability to upvote and optionally downvote discussions.
 
 ### Ideas Overview
```
Should be h2

Charrondev replied:
Ah right. I've removed that h2 (because it's also the h1 at the top) and made all of the h3's h2's now.

slafleche commented on content/help/integrations/wordpress/index.md:
```diff
@@ -1,30 +0,0 @@
-title: WordPress Integration
-tags:
-- Features
-- WordPress
```
Missing tags in merged page

Charrondev replied:
Fixed

slafleche commented on content/help/integrations/wordpress/using-comments.md:
```diff
@@ -1,23 +0,0 @@
-  help:
-    parent: wordpress
-aliases:
-- /features/wordpress/using-comments
```
Missing alias, you also need `/help/integrations/wordpress/using-comments/`

Charrondev replied:
Fixed

slafleche commented on content/help/integrations/wordpress/wishlist.md:
```diff
@@ -1,12 +0,0 @@
----
-title: WishList Integration
```
Missing redirects for this page, don't forget old redirect.

Charrondev replied:
Fixed

DaazKu commented:
Make sure to update the plugins documentation links.

There are one or 2 places where those links are used. For example: https://github.com/vanilla/internal/blob/650e48350b9e627379e4fc115f037c0a4a38d894/plugins/github/addon.json#L12
and https://github.com/vanilla/internal/blob/650e48350b9e627379e4fc115f037c0a4a38d894/plugins/github/views/dashboard.php#L7

Charrondev commented on content/developer/embedding/index.md:
```diff
@@ -3,46 +3,125 @@ title: Embedding
-    identifier: embedding
+    identifier: embedding-developer
 aliases:
-- /features/embedding
```
fixed

Charrondev commented:
I've fixed all of the removed tags and aliases that @slafleche found and implemented his feedback. Additionally I used the new regression test #204 to find a couple more broken links and missing tags. I've opened 2 issues in the relavent repos to update the social connect links (to point to their new subpage pieces.)
